OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs

```diff
@@ -1,48 +1,64 @@
-﻿using Microsoft.AspNetCore.Mvc;
-using Microsoft.Extensions.DependencyInjection;
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Moq;
+using OnlineFlightSearchAPI;
 using OnlineFlightSearchAPI.Controllers;
 using OnlineFlightSearchAPI.FlightServices;
-using OnlineFlightSearchAPI.Repositories;
-using OnlineFlightSearchAPI.Repositories.FlightRepository;
-using System;
-using System.Net;
+using OnlineFlightSearchAPI.Models;
 using Xunit;
 
 namespace OnlineFlightSearchAPITestCases.ControllerTests
 {
-    public class SearchFlightsControllerTests
+    public class SearchFlightsControllerTests : IClassFixture<WebApplicationFactory<OnlineFlightSearchAPI.Startup>>
     {
-        private readonly SearchFlightsController searchFlightsController;
+        private readonly TestServer server;
 
         public SearchFlightsControllerTests()
         {
-            var services = new ServiceCollection();
-            services.AddScoped<SearchFlightsController>();
-            services.AddScoped<ISearchFlightService, FlightService>();
-            services.AddScoped<IAirportServices, AirportServices>();
-            services.AddScoped<IFlightRepository, FlightRepository>();
-            services.AddScoped<IAirportRepository, AirportRepository>();
-            var serviceProvider = services.BuildServiceProvider();
-
-            searchFlightsController = serviceProvider.GetService<SearchFlightsController>();
+            server = new TestServer(new WebHostBuilder()
+                .UseStartup<Startup>());
+        }
+
+        [Theory]
+        [InlineData("BUD", "LTN", "2018-12-12")]
+        public async Task SearchFlightDetails_ValidRequest_ReturnsHttpStatusOK200(string startLocation, string destination, string departureDate)
+        {
+            var client = server.CreateClient();
+            var url = $"/api/SearchFlights/SearchFlightDetails?startLocation={startLocation}&endDestination={destination}&departureDate={departureDate}";
```

Comment:
better url design?
may be /api/searchflightdetails?

Reply:
Yep. URL Redesigned as suggested